metron-analytics/metron-profiler-client/src/test/java/org/apache/metron/profiler/client/HBaseProfilerClientTest.java

```diff
@@ -20,184 +20,187 @@
+  private static final String profileName = "profile1";
+  private static final String entityName = "entity1";
+  private static final int profileValue = 1231121;
+  private static final byte[] profileValueB = SerDeUtils.toBytes(profileValue);
+  private long periodDurationMillis = periodUnits.toMillis(periodDuration);
+
+  private HBaseClient hbaseClient;
+  private HBaseProfilerClient profilerClient;
+  private ProfileMeasurement expected;
+  private RowKeyBuilder rowKeyBuilder;
+  private ColumnBuilder columnBuilder;
+  private Result expectedResult;
+  private Result emptyResult;
 
   @Before
-  public void setup() throws Exception {
-    table = new MockHTable(tableName, columnFamily);
-    executor = new DefaultStellarStatefulExecutor();
-
-    // writes values to be read during testing
-    long periodDurationMillis = periodUnits.toMillis(periodDuration);
-    RowKeyBuilder rowKeyBuilder = new SaltyRowKeyBuilder();
-    ColumnBuilder columnBuilder = new ValueOnlyColumnBuilder(columnFamily);
-    profileWriter = new ProfileWriter(rowKeyBuilder, columnBuilder, table, periodDurationMillis);
-
-    client = new HBaseProfilerClient(table, rowKeyBuilder, columnBuilder, periodDurationMillis);
+  public void setup() {
+    // create a profile measurement used in the tests
+    expected = new ProfileMeasurement()
+            .withProfileName(profileName)
+            .withEntity(entityName)
+            .withPeriod(System.currentTimeMillis(), 5, TimeUnit.MINUTES)
+            .withProfileValue(profileValue);
+
+    // mock row key builder needs to return a row key for the profile measurement used in the tests
+    rowKeyBuilder = mock(RowKeyBuilder.class);
+    when(rowKeyBuilder.rowKey(any())).thenReturn(expectedRowKey);
+
+    // mock column builder - column family/qualifier comes from the column builder
+    columnBuilder = mock(ColumnBuilder.class);
+    when(columnBuilder.getColumnFamily()).thenReturn(columnFamily);
+    when(columnBuilder.getColumnQualifier(eq("value"))).thenReturn(columnQualifier);
+
+    // this mock is used to feed data to the profiler client while testing
+    hbaseClient = mock(HBaseTableClient.class);
+
+    // a result that matches the expected profile measurement that can be return by the mock hbase client
+    expectedResult = mock(Result.class);
+    when(expectedResult.containsColumn(eq(columnFamilyB), eq(columnQualifier))).thenReturn(true);
+    when(expectedResult.getValue(eq(columnFamilyB), eq(columnQualifier))).thenReturn(profileValueB);
+
+    // an empty result to use in the tests
+    emptyResult = mock(Result.class);
+    when(emptyResult.containsColumn(any(), any())).thenReturn(false);
+
+    // create the profiler client that will be tested
+    profilerClient = new HBaseProfilerClient(hbaseClient, rowKeyBuilder, columnBuilder, periodDurationMillis);
   }
 
-  @After
-  public void tearDown() throws Exception {
-    table.clear();
+  @Test
```

Comment:
These tests needed updated to continue to function. I re-did these tests because previously the test cases were focused on testing logic that does not actually live in the `HBaseProfilerClient`.  These tests now focus on exclusively testing the logic in `HBaseProfilerClient`, rather than logic performed elsewhere; like in the `ColumnBuilder` for instance.